source/m3_compile.c

```diff
@@ -398,17 +398,21 @@ M3Result  AllocateConstantSlots  (IM3Compilation o, u16 * o_slot, u8 i_type)
 // TOQUE: this usage count system could be eliminated. real world code doesn't frequently trigger it.  just copy to multiple
 // unique slots.
 static inline
-M3Result  IncrementSlotUsageCount  (IM3Compilation o, u16 i_slot)
+M3Result  IncrementSlotUsageCount  (IM3Compilation o, u16 i_slot, u8 i_type)
 {                                                                                       d_m3Assert (i_slot < d_m3MaxFunctionSlots);
     M3Result result = m3Err_none;                                                       d_m3Assert (o->m3Slots [i_slot] > 0);
 
     // OPTZ (memory): 'm3Slots' could still be fused with 'typeStack' if 4 bits were used to indicate: [0,1,2,many]. The many-case
     // would scan 'wasmStack' to determine the actual usage count
-    if (o->m3Slots [i_slot] < 0xFF)
+    for (u16 i = 0; i < GetTypeNumSlots (i_type); ++i, ++i_slot)
```

Comment:
I'd move the `GetTypeNumSlots` earlier, like what other functions do.

(It's a small function and it will be either 1 or 2, so no big deal.)